elasticetcd doesn't start with external etcd process

I wanted to side-step elasticetcd to get a two node cluster with etcd running as a single node external process. And I have trouble testing out the `--no-embed` option.

**Started external etcd**:

```sh
$ etcd --name gluster --initial-advertise-peer-urls http://127.0.0.1:2380 \
>   --listen-peer-urls http://127.0.0.1:2380 \
>   --listen-client-urls http://127.0.0.1:2379  \
>   --advertise-client-urls http://127.0.0.1:2379 \
>   --initial-cluster gluster=http://127.0.0.1:2380 \
>   --initial-cluster-state new
2017-07-26 17:30:56.821224 I | etcdmain: etcd Version: 3.1.0-rc.0+git
2017-07-26 17:30:56.821644 I | etcdmain: Git SHA: Not provided (use ./build instead of go build)
2017-07-26 17:30:56.821848 I | etcdmain: Go Version: go1.8
2017-07-26 17:30:56.822033 I | etcdmain: Go OS/Arch: linux/amd64
2017-07-26 17:30:56.822231 I | etcdmain: setting maximum number of CPUs to 1, total number of available CPUs is 1
2017-07-26 17:30:56.822414 W | etcdmain: no data-dir provided, using default data-dir ./gluster.etcd
2017-07-26 17:30:56.822660 I | embed: listening for peers on http://127.0.0.1:2380
2017-07-26 17:30:56.822871 I | embed: listening for client requests on 127.0.0.1:2379
2017-07-26 17:30:56.827854 I | etcdserver: name = gluster
2017-07-26 17:30:56.828117 I | etcdserver: data dir = gluster.etcd
2017-07-26 17:30:56.828296 I | etcdserver: member dir = gluster.etcd/member
2017-07-26 17:30:56.828483 I | etcdserver: heartbeat = 100ms
2017-07-26 17:30:56.828591 I | etcdserver: election = 1000ms
2017-07-26 17:30:56.828743 I | etcdserver: snapshot count = 10000
2017-07-26 17:30:56.828929 I | etcdserver: advertise client URLs = http://127.0.0.1:2379
2017-07-26 17:30:56.829128 I | etcdserver: initial advertise peer URLs = http://127.0.0.1:2380
2017-07-26 17:30:56.829308 I | etcdserver: initial cluster = gluster=http://127.0.0.1:2380
2017-07-26 17:30:56.833049 I | etcdserver: starting member b71f75320dc06a6c in cluster 1c45a069f3a1d796
2017-07-26 17:30:56.833718 I | raft: b71f75320dc06a6c became follower at term 0
2017-07-26 17:30:56.833954 I | raft: newRaft b71f75320dc06a6c [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2017-07-26 17:30:56.834162 I | raft: b71f75320dc06a6c became follower at term 1
2017-07-26 17:30:56.840011 I | etcdserver: starting server... [version: 3.1.0-rc.0+git, cluster version: to_be_decided]
2017-07-26 17:30:56.846500 I | etcdserver/membership: added member b71f75320dc06a6c [http://127.0.0.1:2380] to cluster 1c45a069f3a1d796
2017-07-26 17:30:57.535964 I | raft: b71f75320dc06a6c is starting a new election at term 1
2017-07-26 17:30:57.538049 I | raft: b71f75320dc06a6c became candidate at term 2
2017-07-26 17:30:57.539487 I | raft: b71f75320dc06a6c received MsgVoteResp from b71f75320dc06a6c at term 2
2017-07-26 17:30:57.540320 I | raft: b71f75320dc06a6c became leader at term 2
2017-07-26 17:30:57.540670 I | raft: raft.node: b71f75320dc06a6c elected leader b71f75320dc06a6c at term 2
2017-07-26 17:30:57.541429 I | etcdserver: setting up the initial cluster version to 3.1
2017-07-26 17:30:57.542961 N | etcdserver/membership: set the initial cluster version to 3.1
2017-07-26 17:30:57.543871 I | etcdserver/api: enabled capabilities for version 3.1
2017-07-26 17:30:57.544285 E | etcdmain: forgot to set Type=notify in systemd service file?
2017-07-26 17:30:57.544547 I | embed: ready to serve client requests
2017-07-26 17:30:57.554047 N | embed: serving insecure client requests on 127.0.0.1:2379, this is strongly discouraged!
2017-07-26 17:30:57.555256 I | etcdserver: published {Name:gluster ClientURLs:[http://127.0.0.1:2379]} to cluster 1c45a069f3a1d796
```

**Start glusterd2**: 

```sh
# cat noembed.yaml 
workdir: "/home/ppai/workdir/1"
peeraddress: "127.0.0.1:24008"
clientaddress: "127.0.0.1:24007"
no-embed: true
etcdendpoints: "http://127.0.0.1:2379"
```

```sh
[root@gd2-1 workdir]# glusterd2 --config noembed.yaml 
DEBU[2017-07-26T17:35:23+05:30] Starting GlusterD                             pid=26096 version=v4.0dev-7.12.git25d0a01
INFO[2017-07-26T17:35:23+05:30] loaded configuration from file                file=noembed.yaml
DEBU[2017-07-26T17:35:23+05:30] running with configuration                    clientaddress=127.0.0.1:24007 config=noembed.yaml defaultpeerport=24008 etcdcurls= etcdendpoints=http://127.0.0.1:2379 etcdpurls= localstatedir=/home/ppai/workdir/1 logdir=/home/ppai/workdir/1/log logfile=- loglevel=debug mgmt=false no-embed=true peeraddress=127.0.0.1:24008 rundir=/home/ppai/workdir/1/run version=false workdir=/home/ppai/workdir/1
INFO[2017-07-26T17:35:23+05:30] Generated new UUID                            uuid=4b05d159-59df-4652-b953-15b12c2c386d
WARN[2017-07-26T17:35:23+05:30] could not read store config file, continuing with defaults  error=open /home/ppai/workdir/1/store.toml: no such file or directory
DEBU[2017-07-26T17:35:23+05:30] saving updated store config                  
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/gluster/glusterd2/store.newRemoteStore(0xc42043c620, 0xcedc07, 0xc4203d1c68, 0xc4203d1c60)
	/home/ppai/work/src/github.com/gluster/glusterd2/store/remote.go:14 +0x568
github.com/gluster/glusterd2/store.New(0x0, 0x11cfa48, 0x1d58258, 0xc4201e10b0)
	/home/ppai/work/src/github.com/gluster/glusterd2/store/store.go:86 +0x44
github.com/gluster/glusterd2/store.Init(0x0, 0x0, 0x0)
	/home/ppai/work/src/github.com/gluster/glusterd2/store/store.go:52 +0x89
main.main()
	/home/ppai/work/src/github.com/gluster/glusterd2/main.go:79 +0x4c1
```

--- a/glide.lock
+++ b/glide.lock
@@ -1,5 +1,5 @@
-hash: 13a2b1d387fd998c460b8ff818310635acc3254404ffc38a291c536b7acee7c6
-updated: 2017-06-16T13:57:33.232246104+05:30
+hash: a57234017317f401adaca55e4ed11dc6d9b18107b7950e9f55e4efacb4825d17
+updated: 2017-07-28T12:17:49.200749485+05:30
 imports:
 - name: github.com/beorn7/perks
   version: 4c0e84591b9aa9e6dcfdf3e020114cd81f89d5f9
@@ -10,7 +10,7 @@ imports:
 - name: github.com/cockroachdb/cmux
   version: 112f0506e7743d64a6eb8fedbcff13d9979bbf92
 - name: github.com/coreos/etcd
-  version: 66722b1ada68fcd5227db853ee92003169a975c8
+  version: c31bec0f29facff13f7c3e3d948e55dd6689ed42
   subpackages:
   - alarm
   - auth
@@ -85,7 +85,6 @@ imports:
 - name: github.com/coreos/go-systemd
   version: 48702e0da86bd25e76cfef347e2adeb434a0d0a6
   subpackages:
-  - daemon
   - dbus
   - journal
   - unit
@@ -183,7 +182,7 @@ imports:
 - name: github.com/prashanthpai/sunrpc
   version: 8b1e9b0138a37e93705639161d846baed18290c7
 - name: github.com/prometheus/client_golang
-  version: de4d4ffe63b9eff7f27484fdef6e421597e6abb4
+  version: 42552c195dd3f3089fbf9cf26e139da150af35aa
   subpackages:
   - prometheus
   - prometheus/promhttp
@@ -249,6 +248,7 @@ imports:
   version: 69483b4bd14f5845b5a1e55bca19e954e827f1d0
   subpackages:
   - assert
+  - require
 - name: github.com/thejerf/suture
   version: 0ac47afae95ad5bc5184ed346bc945168e883f5d
 - name: github.com/ugorji/go
@@ -260,7 +260,7 @@ imports:
 - name: github.com/xiang90/probing
   version: 07dd2e8dfe18522e9c447ba95f2fe95262f63bb2
 - name: golang.org/x/crypto
-  version: 850760c427c516be930bc91280636328f1a62286
+  version: 7e9105388ebff089b3f99f0ef676ea55a6da3a7e
   subpackages:
   - bcrypt
   - blowfish

--- a/glide.yaml
+++ b/glide.yaml
@@ -1,7 +1,7 @@
 package: github.com/gluster/glusterd2
 import:
 - package: github.com/coreos/etcd
-  version: ~v3.2.0
+  version: ~v3.2.4
   subpackages:
   - client
   - clientv3

--- a/store/remote.go
+++ b/store/remote.go
@@ -3,17 +3,12 @@ package store
 import (
 	"time"
 
-	"github.com/gluster/glusterd2/pkg/elasticetcd"
-
 	log "github.com/Sirupsen/logrus"
 	"github.com/coreos/etcd/clientv3"
 	"github.com/coreos/etcd/clientv3/concurrency"
 )
 
 func newRemoteStore(conf *Config) (*GDStore, error) {
-	if len(conf.Endpoints) == 1 && conf.Endpoints[1] == elasticetcd.DefaultEndpoint {
-		return nil, ErrEndpointsRequired
-	}
 
 	c, e := clientv3.New(clientv3.Config{
 		Endpoints:        conf.Endpoints,

--- a/store/store.go
+++ b/store/store.go
@@ -23,8 +23,6 @@ var (
 	Store *GDStore
 	lock  sync.Mutex
 
-	// ErrEndpointsRequired is returned when endpoints are required but none are given
-	ErrEndpointsRequired = errors.New("etcd endpoints for remote etcd cluster required with no-embed")
 	// ErrStoreInitedAlready is returned when the store is already intialized
 	ErrStoreInitedAlready = errors.New("store has been intialized already")
 )

